components/series/stackedBar.js

```diff
@@ -4,79 +4,43 @@
     fc.series.stackedBar = function() {
 
         var stackedBar = function(selection) {
-            var container;
-
-            // takes an object with values associated with each property, and
-            // converts it into an array of values. Each value has the xValue associated
-            // with it.
-            //
-            // For example, this object:
-            //
-            // obj = { county: 'North Tyneside', labour: 23, conservative: 55 }
-            //
-            // becomes this ...
-            //
-            // [
-            //   { xValue: 'North Tyneside', name: 'labour', previousValue: 0, currentValue: 23},
-            //   { xValue: 'North Tyneside', name: 'conservative', previousValue: 23, currentValue: 78},
-            // ]
-            function objectDatapointToArray(obj) {
-                var values = [];
-                var yTotal = 0;
-                var xVal = obj[stackedBar.xValueKey.value];
-                for (var propertyName in obj) {
-                    if (obj.hasOwnProperty(propertyName) && propertyName !== stackedBar.xValueKey.value) {
-                        var previous = yTotal;
-                        yTotal += Number(obj[propertyName]);
-                        values.push({
-                            'name': propertyName,
-                            'previousValue': previous,
-                            'currentValue': yTotal,
-                            'xValue': xVal
-                        });
-                    }
-                }
-                return values;
-            }
 
             selection.each(function(data) {
 
                 // add a 'root' g element on the first enter selection. This ensures
                 // that it is just added once
-                container = d3.select(this);
+                var container = d3.select(this);
 
-                var keyFunction = function(d) {
-                    return d[stackedBar.xValueKey.value];
-                };
-                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', data, keyFunction);
+                var layers = stackLayout(data);
 
+                var g = fc.utilities.simpleDataJoin(container, 'stacked-bar', layers);
 
-                // create a join for each bar
                 var bar = g.selectAll('rect')
-                    .data(function(d) { return objectDatapointToArray(d); })
+                    .data(function(d) { return stackLayout.values()(d); })
```

Comment:
Sorry, I wasn't clear in my previous feedback and this yet again points to why we should remove this pattern.

Within the component declaring a property, our convention has been to reference the value using the ```.value``` member. Outside of the component, references are made by invoking the function with zero-args. 

So in this case it would be ```data(function(d) { return stackLayout.values.value(d); })```, which would then simplify to ```data(stackLayout.values.value)```.